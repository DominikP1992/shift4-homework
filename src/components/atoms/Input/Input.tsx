import theme from '@/styles/theme';
import styled from '@xstyled/emotion';

const Input = styled.input`
  border-radius: 4px;
  padding: 0 16px;
  height: ${theme.sizes.xl};
  font-size: 24px;
  font-weight: 500;
  font-family: Rubik;
  background-color: transparent;
  color: ${theme.colors['purple-gray']};
  border: 1px solid ${theme.colors['input-border']};
  outline: none;
  box-shadow: none;
  ${({ fullWidth }: { fullWidth?: boolean }) => fullWidth && `width:100%;`}

  &:hover {
    box-shadow: 0 0 0 1px ${theme.colors['icon-button--pressed']};
  }

  &:focus {
    box-shadow: 0 0 0 1px ${theme.colors['input-border--active']};
  }

  &::placeholder {
    color: ${theme.colors.gray};
  }
`;

export default Input;
