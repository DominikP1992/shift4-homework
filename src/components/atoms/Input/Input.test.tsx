import React from 'react';
import { render, screen } from '@testing-library/react';
import Input from './Input'; // Adjust the import path as necessary

describe('Input', () => {
  it('renders without crashing', () => {
    render(<Input aria-label="test input" />);
    expect(screen.getByLabelText('test input')).toBeInTheDocument();
  });

  it('match snapshot', () => {
    render(<Input aria-label="test input" />);
    expect(screen.getByLabelText('test input')).toMatchSnapshot();
  });

  it('spans full width when fullWidth is true', () => {
    render(<Input fullWidth aria-label="test input" />);

    expect(screen.getByLabelText('test input')).toHaveStyle('width: 100%');
  });
});
