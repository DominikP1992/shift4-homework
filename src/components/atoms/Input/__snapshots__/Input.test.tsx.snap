// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input match snapshot 1`] = `
<input
  aria-label="test input"
  class="css-185unua"
/>
`;
